admin_solutions/master_node_config.adoc

```diff
@@ -0,0 +1,409 @@
+
+----
+$ ansible-playbook -b -i ./hosts ~/src/openshift-ansible/playbooks/byo/openshift-cluster/config.yml
+----
+
+To modify the Ansible inventory and make configuration changes:
+
+. Open the inventory file:
++
+.Sample inventory file:
+====
+
+$ atom ~/src/openshift-ansible/playbooks/byo/openshift-cluster/config.yml
+----
+[OSEv3:children]
+masters
+nodes
+
+[OSEv3:vars]
+ansible_ssh_user=cloud-user
+ansible_become=true
+#openshift_image_tag=v3.1.1.6
+#openshift_image_tag=v3.2.0.14
+deployment_type=openshift-enterprise
+
+openshift_docker_additional_registries=ec2-54-172-104-209.compute-1.amazonaws.com:5000
+openshift_docker_insecure_registries=ec2-54-172-104-209.compute-1.amazonaws.com:5000
+
+[masters]
+ec2-54-173-187-15.compute-1.amazonaws.com  openshift_ip=172.18.4.89 openshift_public_ip=54.173.187.15 openshift_hostname=ip-172-18-4-89.ec2.internal openshift_public_hostname=ec2-54-173-187-15.compute-1.amazonaws.com containerized=True
+[nodes]
+ec2-54-173-187-15.compute-1.amazonaws.com  openshift_ip=172.18.4.89 openshift_public_ip=54.173.187.15 openshift_hostname=ip-172-18-4-89.ec2.internal openshift_public_hostname=ec2-54-173-187-15.compute-1.amazonaws.com containerized=True openshift_schedulable=False
+ec2-54-89-5-36.compute-1.amazonaws.com  openshift_ip=172.18.4.90 openshift_public_ip=54.89.5.36 openshift_hostname=ip-172-18-4-90.ec2.internal openshift_public_hostname=ec2-54-89-5-36.compute-1.amazonaws.com containerized=True
+
+[nodes]
+ec2-54-173-187-15.compute-1.amazonaws.com  openshift_ip=172.18.4.89 openshift_public_ip=54.173.187.15 openshift_hostname=ip-172-18-4-89.ec2.internal openshift_public_hostname=ec2-54-173-187-15.compute-1.amazonaws.com containerized=True openshift_schedulable=False
+ec2-54-89-5-36.compute-1.amazonaws.com  openshift_ip=172.18.4.90 openshift_public_ip=54.89.5.36 openshift_hostname=ip-172-18-4-90.ec2.internal openshift_public_hostname=ec2-54-89-5-36.compute-1.amazonaws.com containerized=True
+----
+====
++
+. Add the following new variables to the `[OSEv3:vars]` section of the file:
++
+----
+# htpasswd auth
+openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]
+# Defining htpasswd users
+openshift_master_htpasswd_users={'jsmith': '$apr1$wIwXkFLI$bAygtISk2eKGmPOqaJftB', 'bloblaw': '7IRJ$2ODmeLoxf4I6sUEKfiA$2aeKnsDJqLJe'
+# or
+#openshift_master_htpasswd_file=<path/to/local/pre-generated/htpasswdfile>
+----
++
+This will change the authentication method from the default `deny all` setting to `htpasswd` and use the specified file to generate user IDs and passwords for `jsmith` and `bloblaw`.
+. Re-run the ansible playbook for these modifications to take effect:
++
+----
+$ ansible-playbook -b -i ./hosts ~/src/openshift-ansible/playbooks/byo/openshift-cluster/config.yml
```

Comment:
OK, thanks. Updated to:

$ ansible-playbook -b -i ./hosts ~/src/openshift-ansible/playbooks/byo/config.yml